core/rawdb/accessors_chain.go

```diff
@@ -299,39 +302,114 @@ func ReadReceiptsRLP(db ethdb.Reader, hash common.Hash, number uint64) rlp.RawVa
 	return data
 }
 
-// ReadReceipts retrieves all the transaction receipts belonging to a block.
-func ReadReceipts(db ethdb.Reader, hash common.Hash, number uint64) types.Receipts {
+// ReadRawReceipts retrieves all the transaction receipts belonging to a block.
+// The receipt metadata fields are not guaranteed to be populated, so they
+// should not be used. Use ReadReceipts instead if the metadata is needed.
+func ReadRawReceipts(db ethdb.Reader, hash common.Hash, number uint64) types.Receipts {
 	// Retrieve the flattened receipt slice
 	data := ReadReceiptsRLP(db, hash, number)
 	if len(data) == 0 {
 		return nil
 	}
+
```

Comment:
Nitpick, but general repo style: Could you avoid here (and throughout all the PR) empty lines like this? When you close a previous block `}` that already acts as an empty line/visual separator, so adding one more doesn't aid much, just makes the code fall apart. Very rarely it makes sense to add one more empty line if you really want to emphasize the previous code from the next one, but generally there's no need.

Reply:
@karalabe No problem. I'll ensure the code in this PR adheres to that convention.